src/model/geom/track.py

```diff
@@ -50,3 +51,23 @@ def sense_closest(self, sensor: Sensor, active_segment: SegmentId) -> float:
                 if (d := sensor.check_distance(wall)) is not None:
                     return d
         raise RuntimeError("couldn't find any walls in any of the segments")
+
+    @cached_property
+    def boundaries(self) -> Tuple[Vec2, Vec2]:
+        """:return: Tuple[Vec2, Vec2] - upper left and lower right corners of rectangle enclosing track
+        """
+        first_point = self.segments[0].right_wall.line_segment.points[0]
+        min_x = first_point.x
+        min_y = first_point.y
+        max_x = first_point.x
+        max_y = first_point.y
+        for segment in self.segments:
+            for point in (
+                segment.left_wall.line_segment.points
+                + segment.right_wall.line_segment.points
+            ):
+                min_x = min(min_x, point.x)
+                min_y = min(min_y, point.y)
+                max_x = max(max_x, point.x)
+                max_y = max(max_y, point.y)
+        return Vec2(min_x, min_y), Vec2(max_x, max_y)
```

Comment:
`planar` has a dedicated type for this.
```suggestion
    @cached_property
    def bounding_box(self) -> BoundingBox:
        return BoundingBox.from_shapes(segment.region for segment in self.segments)
```

Reply:
This proposed code fails

Reply:
For some reason `planar.Polygon` isn't iterable when using pure python version, the library has to be setup with C module support.